controllers/nginx/pkg/cmd/controller/tcp.go

```diff
@@ -65,10 +67,31 @@ func (p *proxy) Handle(conn net.Conn) {
 	}
 	defer clientConn.Close()
 
-	_, err = clientConn.Write(data[:length])
+	if proxy.ProxyProtocol {
+		//Write out the proxy-protocol header
+		localAddr := conn.LocalAddr().(*net.TCPAddr)
+		remoteAddr := conn.RemoteAddr().(*net.TCPAddr)
+		protocol := "UNKNOWN"
```

Comment:
I thought about that - I wasn't sure if one of To4() and To16() != nil is guaranteed with the golang net stdlib - the proxy-protocol spec says that if it's not either IPv4 or IPv6 the protocol header should read 'UNKNOWN'.